File: client/src/modules/lessons/models/LessonWriteRequest.ts
import type { LessonStatus } from '@/types/LessonStatus'

export interface LessonWriteRequest {
  title: string
  description: string | null
  contentHtml: string | null
  videoUrl: string | null
  videoDurationSeconds?: number | null
  sortOrder: number
  status: LessonStatus
  version?: number
}

File: server/src/Elearning.Api/Contracts/Lessons/Requests/LessonWriteRequest.cs

using Elearning.Domain;

namespace Elearning.Api.Contracts.Lessons.Requests;

public sealed record LessonWriteRequest(
    string Title,
    string? Description,
    string? ContentHtml,
    string? VideoUrl,
    int SortOrder,
    LessonStatus Status,
    int? VideoDurationSeconds = null,
    long? Version = null);
